### PR TITLE
fix: use cycle position endpoints for favs

### PR DESCRIPTION
**Pull request by rtirserio**
requires https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/92

should be no difference in how this works, its just using a different model behind the scenes

### CODE REVIEW

File: src/Components/CompareList/CompareList.jsx [hunk 1]
```diff
@@ -296,7 +296,7 @@ class CompareList extends Component {
                           <td key={shortId.generate()}>
                             <Favorite
                               hasBorder
-                              refKey={c.position.id}
+                              refKey={c.id}
```
Comment (mjoyce91) [inline src/Components/CompareList/CompareList.jsx]: There's one of these on the position details page that needs to be updated as well

Comment (mjoyce91) [reply src/Components/CompareList/CompareList.jsx]: Example: http://localhost:3000/details/270 - adding favorite fails

Comment (mjoyce91) [reply src/Components/CompareList/CompareList.jsx]: You could do something like this:

src/Components/PositionDetails/PositionDetails.jsx:
```jsx
...
<PositionTitle
  details={{ ...position, cycleId: details.id }}
...
```

src/Components/PositionTitle/PositionTitle.jsx:
```jsx
...
<Favorite
    refKey={details.cycleId}
...
```

Comment (rtirserio) [reply src/Components/CompareList/CompareList.jsx]: updated

**Comment by mjoyce91** [comment]
@rtirserio there was one more place this change had to be added. Approving and will let you merge if it looks good.

**Comment by rtirserio** [comment]
@mjoyce91 we also need https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/92

**Comment by mjoyce91** [comment]
Approved

**Pull request merged**